games/static/games/proto.js

```diff
@@ -307,87 +347,146 @@ class Board {
         return !this.squares[this.defenderLocation].isRightBorder()
     }
 
-    shootRocket(e) {
+    shootRocket() {
         let rocketId
-        const squareAboveDefender = this.squares[this.defenderLocation-this.width]
-        let rocket = new Rocket('rocket', squareAboveDefender, this)
-        squareAboveDefender.setCharacter(rocket)
-        console.log(rocket)
-        function moveRocket(rocket) {
-            console.log(rocket)
-            let location = rocket.indexOnBoard
-            let rocketInFlight = rocket.indexOnboard.pickCharacterUp()
-            let newLocation = location - this.width
-            try {
-                this.squares[newLocation].setCharacter(rocketInFlight)
+        let rocket = this.createRocket()
+        let rocketLocation = this.getIndexOf(rocket)
+        let nextRocketLocation = this.getNextIndexOf(rocket)
+        this.squareAt(rocketLocation).setCharacter(rocket)
+        rocketId = setInterval(() => {
+            let rocketInFlight = this.liftRocketAt(rocketLocation)
+            if (rocketInFlight.isAlien()) {
+                this.squareAt(rocketLocation).setCharacter(rocket)
+                rocketInFlight = this.liftRocketAt(rocketLocation)
```

Comment:
why did you lift the rocket, put it down and lift it again?

Reply:
ther was a bug where the rocket would accidentally turn to an alien and drag it up the board, this checks for this condition and if its true it turns it back to a rocket

Reply:
ok so please explain in comments